### ISSUE TITLE
Problems using other base learners other than DecisionTreeRegressor

### ISSUE THREAD

**Issue by henriqueluzz** [stanfordmlgroup/ngboost#172]
Hello,

I'm trying to use others base learners other than scikit-learn DecisionTreeRegressor(), I tried RandomForestRegressor() and GradientTreeRegressor(), since as far as I understood the base regressor doesn't need to be probabilistic.

Both of them when I fit the model, the training hangs forever and doesn't output any messages. 
Not sure if I'm missing something from the ngboost fundamentals or it's not working properly with other base learners.

```
# Base Regressor
from sklearn.tree import DecisionTreeRegressor
from sklearn.ensemble import GradientBoostingRegressor, RandomForestRegressor

#base_regressor = RandomForestRegressor()
#base_regressor = GradientBoostingRegressor()
base_regressor = DecisionTreeRegressor()

# Set max_depth to 15 basically.
base_regressor.set_params(**params['regressor_params'])

ngb = NGBRegressor(Base=base_regressor,
                   Dist=distribution[params['distribution']],
                   learning_rate=params['learning_rate'],
                   n_estimators=params['n_estimators']).fit(X_train_full, y_train_full,
                                                            X_val=X_valid, Y_val=y_valid,
                                                            early_stopping_rounds=5)
```

Thanks in advance.

**Comment by ryan-wolbeck** [comment]
@henriqueluzz Whats the shape of your input dataframe? Also would like to see what regressor_params to set to?

**Comment by henriqueluzz** [comment]
@ryan-wolbeck 
The params dict
```
params =  {"learning_rate": 0.01,
           "n_estimators":  3000,
           "distribution": "Normal",
           "regressor_params": {"max_depth": 18}}
```
X_train_full has (370410, 31) shape

**Comment by ryan-wolbeck** [comment]
@henriqueluzz 

I think it's probably running fine but with those params I'm guessing that it's taking a long time for each iteration, that learning rate is pretty small, n_estimators is really high and the depth is pretty substantial given your sample. I would loosen those parameters  and set verbose eval to a lower number than the default 100
https://github.com/stanfordmlgroup/ngboost/blob/e1a95a15fb95a358b72fd6d76d212a5a66e80dee/ngboost/api.py#L52

**Comment by alejandroschuler** [comment]
Yeah I agree- your data is pretty big and running with depth=18 is pretty crazy. Typically depth <5 is used in boosting. Using full-fledged models like random forests as base learners is theoretically possible but not necessarily advisable. It will take a long time to fit each boosting stage and the result may not necessarily be much better, if it is better at all.

**Comment by henriqueluzz** [comment]
I will loosen the parameters and give it a try, although these hyperparams came from a LightGBM regression model using the same data that ran an extensive hyperparameter search using hyperopt. 
But that doesn't guarantee to be suitable for other boosting implementations.

Thanks for the responses, will update since I get the results
